architecture/2024-08-bicepdeployment-controller.md

```diff
@@ -0,0 +1,379 @@
+# Radius BicepDeployment Controller
+
+* **Author**: Will Smith (@willdavsmith)
+
+## Overview
+
+Today, users of Radius and future adopters of Radius use many Kubernetes-specific tools in their production workflows. Most of these tools operate on Kubernetes resources exclusively - which presents a problem when trying to deploy resources defined in Bicep manifests. This design proposes the creation of a new Kubernetes reconciler in Radius called the BicepDeployment Reconciler that will allow users to deploy resources defined in Bicep manifests using Kubernetes tooling.
+
```

Comment:
And for a point of comparison https://fluxcd.io/blog/2022/09/how-to-gitops-your-terraform/ exists.

Running terraform once lets you create something. Using Flux to run Terraform let's you continually apply the same state. This is what we're enabling but with Bicep.